codegen/module.go

```diff
@@ -557,93 +557,87 @@ func (system *ModuleSystem) ResolveModules(
+				if err != nil {
+					return nil, errors.Wrapf(
+						err,
+						"internal error: cannot stat %q",
+						moduleDirAbs,
+					)
+				}
 
-			instance, instanceErr := system.readInstance(
-				packageRoot,
-				baseDirectory,
-				targetGenDir,
-				moduleDir,
-			)
-			if instanceErr != nil {
-				if _, ok := instanceErr.(inferError); ok {
-					// It could be the case that a glob pattern like "endpoints/*" matches a *folder* that is later
-					// matched by a longer glob pattern like "endpoints/tchannel/*". It will fail at a later step
-					// if no globbing pattern matches the wanted instance.
+				if !stat.IsDir() {
+					// If a *-config.yaml file, or any other metadata file also matched the glob, skip it, since we are
+					// interested only in the containing directories.
 					continue
-				} else {
+				}
+
+				moduleDir, err := filepath.Rel(baseDirectory, moduleDirAbs)
+				if err != nil {
 					return nil, errors.Wrapf(
-						instanceErr,
-						"Error reading multi instance %q",
-						moduleDir,
+						err,
+						"internal error: cannot make %q relative to %q",
+						moduleDirAbs,
+						baseDirectory,
 					)
 				}
-			}
-
-			resolvedModules[instance.ClassName] = append(resolvedModules[instance.ClassName], instance)
 
-			// For consistency with prior zanzibar behavior of assuming all instances in a multi-module directory
-			// are of the same class name, we will return an error in this case.
-			if className == "" {
-				className = instance.ClassName
-			} else if className != instance.ClassName {
-				return nil, fmt.Errorf(
-					"invariant: all instances in a multi-module directory %q are of type %q (violated by %q)",
-					moduleDirectoryGlob,
-					className,
+				instance, instanceErr := system.readInstance(
+					packageRoot,
+					baseDirectory,
+					targetGenDir,
 					moduleDir,
 				)
+				if instanceErr != nil {
+					if _, ok := instanceErr.(inferError); ok {
+						// It could be the case that a glob pattern like "endpoints/*" matches a *folder* that is later
+						// matched by a longer glob pattern like "endpoints/tchannel/*". It will fail at a later step
+						// if no globbing pattern matches the wanted instance.
+						continue
+					} else {
+						return nil, errors.Wrapf(
+							instanceErr,
+							"Error reading multi instance %q",
+							moduleDir,
+						)
+					}
+				}
+
+				resolvedModules[instance.ClassName] = append(resolvedModules[instance.ClassName], instance)
+
+				if className != instance.ClassName {
+					return nil, fmt.Errorf(
+						"invariant: all instances in a multi-module directory %q are of type %q (violated by %q)",
+						moduleDirectoryGlob,
+						className,
+						moduleDir,
+					)
+				}
 			}
-		}
 
-		// Resolve dependencies for all classes
-		resolveErr := system.populateResolvedDependencies(
-			resolvedModules[className],
-			resolvedModules,
-		)
-		if resolveErr != nil {
-			return nil, resolveErr
-		}
+			// Resolve dependencies for all classes
```

Comment:
Are these tabs? gofmt didn't taken care of it?

Reply:
Yes they're tabs. They look improperly indented in this pull request but they're properly indented in code.